train.py

```diff
@@ -153,25 +161,73 @@ def train(data_path: str, tb_path: str = None, n_scenes: int = 85, version: str
     dataset = NuscenesBEVDataset(nuscenes=nuscenes, n_scenes=n_scenes)
     train_loader = DataLoader(dataset, batch_size=batch_size, shuffle=True, num_workers=n_loader_workers,
                               collate_fn=frames_bboxes_collate_fn, pin_memory=True)
+    # TODO: create actual validation data-loader
+    val_loader = DataLoader(dataset, batch_size=batch_size, shuffle=True, num_workers=n_loader_workers,
+                            collate_fn=frames_bboxes_collate_fn, pin_memory=True)
     print('Loaders are ready.\n',
-          f'Number of batches in train loader: {len(train_loader)}\n')
-          # f'Number of bathces in validation loader: {len(val_loader)}')
+          f'Number of batches in train loader: {len(train_loader)}\n'
+          f'Number of bathces in validation loader: {len(val_loader)}')
 
     frame_depth, frame_width, frame_length = dataset.grid_size
     model = Detector(img_depth=frame_depth).to(device)
     criterion = DetectionLoss()
     optimizer = Adam(model.parameters(), lr=1e-4)
     scheduler = StepLR(optimizer, gamma=0.5, step_size=50)  # TODO: adjust step_size empirically
     detector_out_shape = batch_size, model.out_channels, frame_width // (2 ** model.n_pools), \
-        frame_length // (2 ** model.n_pools)
+                         frame_length // (2 ** model.n_pools)
     gt_former = GroundTruthFormer((frame_width, frame_length), detector_out_shape)
 
+    best_val_loss, best_val_score = None, None
+
     for epoch in trange(n_epochs, desc="Epoch"):
         run_epoch(model, train_loader, criterion, gt_former, epoch, mode='train', writer=train_writer,
                   optimizer=optimizer)
         scheduler.step()
-        # run_epoch(model, val_loader, criterion, epoch, mode='val', writer=val_writer)
+        val_loss, val_score = run_epoch(model, val_loader, criterion, gt_former, epoch, mode='val', writer=val_writer)
+        # saving model weights in case validation loss AND score are better
+        if not best_val_loss or (val_loss < best_val_loss and val_score > best_val_score):
```

Comment:
Compare to None always with `is None`
```suggestion
        if best_val_loss is not None or (val_loss < best_val_loss and val_score > best_val_score):
```